### ISSUE TITLE
Bezels not working & Buttons have strange white boxes

### ISSUE THREAD

**Issue by Rulerofzeworld** [ParthJadhav/Tkinter-Designer#127]
The bezels on the bottom box doesn't seem to be working.
Also, when the button is clicked, it creates a strange white box on the surrounding background.
See images below:
![image](https://user-images.githubusercontent.com/43553166/131902010-88233ee6-932f-492a-9ff7-7f0fc2b86320.png)
![image](https://user-images.githubusercontent.com/43553166/131902087-58930888-3aa0-4acc-bf79-8b255fc4b4ef.png)
![image](https://user-images.githubusercontent.com/43553166/131902202-bfaa42bc-bbc4-4a12-9a61-c2eccc84a0c0.png)

**Comment by jvendegna** [comment]
👀  I see that weird border. From the images you provided it looks to me like you have an extra rectangle in your button group. You can refer to [this example figma file](https://www.figma.com/file/LeDp52QSe7qM8VQLTT5nRD/tkd2PageExample?node-id=0%3A1) for a reference of how buttons must be structured. I can't speak for the bezels without more information. I'm not a designer, just holding down the fort.

**Comment by Rulerofzeworld** [comment]
@jvendegna 
I just ran tkinter-designer on the example figma file you sent me.

Here's the output:
![image](https://user-images.githubusercontent.com/43553166/132098608-c09aaad0-f909-4874-aa08-5f1b161c92ff.png)
Here's the expected result:
![image](https://user-images.githubusercontent.com/43553166/132098629-2e1eacbb-f195-4f0e-8992-553391ceeba5.png)

**Comment by jvendegna** [comment]
Fair enough, I appreciate your input and response. Let's see what, if anything I can do. Parth is out for about another month. The expected result would not include the text for 'Frame 1' but I understand your sentiment. It could be my example is making a similar mistake. I'll start digging and get back to you asap.

**Comment by Rulerofzeworld** [comment]
Awesome, thanks. Looking forward to your response.
The 'Frame 1' text wasn't intended, just accidentally included it in the screenshot.

**Comment by jvendegna** [comment]
I got it. I actually stopped half way through reading [the element guide in the instructions](https://github.com/ParthJadhav/Tkinter-Designer/blob/master/docs/instructions.md#2-element-guide)... Always read the docs.

I _added_ a rectangle layer without rounded corners underneath the rounded rectangle in the button group, then set the background color to the same as the frame background color. Results look a lot better.

![image](https://user-images.githubusercontent.com/58440894/132100525-a57313e9-8657-403f-84fc-3192ed2f0e24.png)

The method I suggest is to simply copy the existing rounded rectangle, paste, remove the rounding from the corners, set the bg color. This way you can be sure they are perfectly aligned.

**Comment by jvendegna** [comment]
Now I think I understand your question better. My design also makes that weird white border when I click the buttons. Must be a glitchy animation or something. I'll keep digging.

**Comment by Rulerofzeworld** [comment]
Awesome thanks.
I just checked the gui that's downloaded from this repository since it's apparently made with figma and imported to tkinter using Tkinter-Designer, and it seems to have the same issue.
![image](https://user-images.githubusercontent.com/43553166/132100753-7b89ce23-194d-48f0-add9-91969ececf3b.png)

Much less noticeable due to the white background, but very visible on a dark background.

**Comment by jvendegna** [comment]
Okay, I have a workaround for you. In your resulting build/gui.py file you'll find a line near the top with:
`window.configure(bg = <hex_color>)`

Copy that, or if you know your hex color just modify the gui.py file where you have buttons, like so:
For example, change this:
```
button_1 = Button(
    image=button_image_1,
    borderwidth=0,
    highlightthickness=0,
    command=lambda: print("button_1 clicked"),
    relief="flat"
)
```
to
```
button_1 = Button(
    image=button_image_1,
    borderwidth=0,
    highlightthickness=0,
    activebackground="<your hex color with the # symbol>",
    command=lambda: print("button_1 clicked"),
    relief="flat"
)
```
I'll push up a fix to the generator. But it'll be a while before it gets merged. Feel free to clone the branch. You'll see it tagged here.

**Comment by Rulerofzeworld** [comment]
The temporary (manual) fix seems to work well 👍 
Not very familiar with the code, but it seems like the frame object passed into the button always has a white background. Might need to set the frame's colour as the same as the canvas on initialisation (for the pull request). Also had to set `self.frame = frame` in the `__init__` function in addition with your fix for it to add `activebackground` to `gui.py`.

As for the bottom rectangle not having the bezels, I searched through the code and it doesn't seem like they're converted to an image like buttons and text entries are, which means they lose any formatting applied to them in Figma.
Might be better to open as a separate issue if this is the case.

**Comment by jvendegna** [comment]
QQ: I see your edited comments. I am curious, your solution seems to work on my end - adding self.frame = frame to the __init__ method. What went wrong for you, if anything, because of this?

**Comment by ParthJadhav** [comment]
Hey @Rulerofzeworld & @jvendegna the bezels on the button when clicked are indicators that the button is clicked and it's something that is implemented by tkinter to make sure that the user understands that the button has been clicked. 

-----

Also how does adding active background property improves the result. I read the comments but didn't seem to understand it quite well. 

----- 

Also when testing new Pull requests or features try using this file -
https://www.figma.com/file/WVLnulVsI177tvnxSdqOUZ/Untitled

it has all the elements necessary.

**Comment by jvendegna** [comment]
@ParthJadhav the animation on a button click remains, so the user does receive visual feedback on-click, however the bezel adopts the color of the background with my changes. I think it's more visually appealing is all. It could be an optional arg with a default value. But I think this may be a fix for eliminating the need for a second rectangle in the background of a rounded-corner button as well. Making the overall user-experience more uniform.

Thanks for the link to the test-file. I'll add a script to the makefile for generating that UI.
